File: 1-two-sum/two-sum.dart
class Solution {
  List<int> twoSum(List<int> nums, int target) {
    for (int i = 0; i < nums.length; i++) {
      for (int j = nums.length - 1; j > i; j--) {
        if (nums[i] + nums[j] == target) {
          return [i, j];
        }
      }
    }
    throw ArgumentError('No two sum solution');
  }
}

File: 1-two-sum/two-sum.go
func twoSum(nums []int, target int) []int {
	myMap := map[int]int{}

	for i := 0; i < len(nums); i++ {
		diff := target - nums[i]
		index, boolean := myMap[diff]
		if boolean {
			return []int{index, i}
		}
		myMap[nums[i]] = i
	}

	return []int{-1, -1}
}